File: java/src/main/java/com/experian/dvs/client/email/validate/DomainType.java
package com.experian.dvs.client.email.validate;

import java.util.HashMap;
import java.util.Map;

public enum DomainType {
    CONSUMER("consumer"),
    BUSINESS("business");

    private final String name;
    // Map of name to EmailDomainType
    private static final Map<String, DomainType> nameToEmailDomainType = new HashMap<>();
    static {
        for (DomainType type : DomainType.values()) {
            nameToEmailDomainType.put(type.name, type);
        }
    }

    DomainType(final String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public static DomainType fromName(final String name) {
        return nameToEmailDomainType.get(name);
    }
}

File: typescript/src/email/validate/domainType.ts
export enum DomainType {
    Consumer = "consumer",
    Business = "business"
}

export function lookupDomainType(str?: string): DomainType | undefined {
    if (str) {
        return Object.keys(DomainType).find(key => DomainType[key as keyof typeof DomainType] === str) as DomainType | undefined;    
    } else {
        return undefined;
    }
}